### DISCUSSION TITLE
Swagger UI crashes for generated openapi.json for recursive schema definitions

### DISCUSSION THREAD

**Discussion by anil-enable** [fastapi/fastapi#10525] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional

from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()


class Message(BaseModel):
    message: str
    # commenting this will work
    details: Optional[list["Message"]] = None


@app.get("/hello", response_model=Message)
def hello() -> Message:
    return Message(message="Hello World")


# source .venv/bin/activate
# uvicorn main.py --port 8020
```

### Description

The shared code snippet has one API route (`/hello`) that returns Message object. The Message schema has recursive association to itself - Message has details which is a list Message(s). 
- In the swagger UI (http://127.0.0.1:8020/docs) when I expand the endpoint `/hello` it gets stuck in loading, and soon after the whole page becomes unresponsive. 
- If I comment the details field in Message, it works
- I did some digging and I found that swagger-ui-dist 5.9.1 released recently is causing this issue, this was ok in previous version 5.9.0.

I expected this to work like before for all released version of swagger-ui-dist

I have 3 questions:
1. Is there something I have missed, that could help me define the message schema correctly so that swagger doesn't have issues with it?
2. Or is there some tweaking required in the generated openapi json?
3. I have used the code snippet given in https://github.com/tiangolo/fastapi/issues/1762#issuecomment-662162612 to monkey patch `swagger_js_url` to `https://cdn.jsdelivr.net/npm/swagger-ui-dist@5.9.0/swagger-ui-bundle.js`. This works for now. But is there any other way to specify the swagger-ui version to be used?

### Operating System

macOS

### FastAPI Version

0.104.0

### Pydantic Version

2.4.2

### Python Version

3.11.4

### Additional Context

I have also added a comment here https://github.com/swagger-api/swagger-ui/issues/9337#issuecomment-1780849318 regarding the same.

**Comment by mvanderlee** [comment]
Alternative workaround provided here: https://github.com/tiangolo/fastapi/discussions/10524

**Comment by YuriiMotov** [comment]
Seems that has been fixed. In current version it doesn't crash
